website/client/src/components/tasks/spells.vue

```diff
@@ -42,7 +42,7 @@
                 :key="key"
                 v-b-popover.hover.auto="skillNotes(skill)"
                 class="col-12 col-md-3"
-                @click="castStart(skill)"
+                @click="!spellDisabled(key) ? castStart(skill) : null"
```

Comment:
@Tressley this line disables the spell from being cast when not necessary but since the disabled state for spells is missing it'll show the popover saying it's not useful anymore but it'll keep the pointer cursor and hover state. Should it be reversed waiting for https://github.com/HabitRPG/habitica/issues/12362 or it can be shipped with this anyway in the meantime?

Reply:
This should be reversed until we're able to remove that conflicting states. However, the disabled state is very simple and should just be setting `opacity: 0.5;`

Reply:
@JalanshMunshi I'd say the easiest thing is to apply the style @Tressley suggested when the spell is disabled but otherwise reverting the changes to @click is fine as well

Reply:
ah, also the cursor should be set to `cursor: default`

Reply:
@paglias Yeah, we can apply the style in this PR if you want. Will it be possible for you to guide me through it? I tried to fix it on my own, but I was unable owing to lesser experience with Vue. 

Do we have to make the change in the css of `.popover` class when the `skillDisabled` is true?

Reply:
@JalanshMunshi The basic way to apply a style dynamically in Vue is to add a line like this to the element:

`:class="{class-name: expression}"`

where `expression` is the condition when the class should be applied, and `class-name` is a CSS class name--you can define it in the `scoped` block further down the .vue file.

So here, you'd add something like

`:class="{'disabled': spellDisabled(key)}"`

to the `div` on line 48, then under `.spell` in the CSS (say, around line 118) add

```
&.disabled {
  opacity: 0.5;
  cursor: default;
}
```

Does that make sense?

Reply:
@SabreCat Yes, it does! Thanks for explaining it so well. The line 
`:class="{'disabled': spellDisabled(key)}"`

was already added in the code at line 50 in `spells.vue`.

Hence, the only thing remaining was to add the specific class named `.disabled`. Please have a look at the change and let me know if we need to change something.

Reply:
The `:class` line was on the wrong element (in fact, it was on an `a` element with no content... who knows how that got in there in the first place 🤔), so it didn't work. Have you been testing your changes locally after updating the code?

I went ahead and added a commit with the correct fixes, as well as a couple of other bits of cleanup I figured I'd do while I was in there, ha.

Reply:
I had the same doubt and since it was already there, I thought of asking first. My browser console showed that the style was applied, but I was not sure whether we had to keep it inside the `a` element as your previous comment said that we had to put it in the `div` element. Went through your commit. I understand the changes now.

Thanks, @SabreCat ! :smiley:

Reply:
Anything else that needs to be changed/added in this PR?

Reply:
Things are looking good to me. @paglias ?

Reply:
left one new comment but the disabled state looks good